lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+module Travis
+  module Build
+    class Script
+      class R < Script
+        DEFAULTS = {
+          # Basic config options
+          cran: 'http://cran.rstudio.com',
+          warnings_are_errors: false,
+          # Dependencies (installed in this order)
+          apt_packages: [],
+          r_binary_packages: [],
+          r_packages: [],
+          bioc_packages: [],
+          r_github_packages: [],
+          # Build/test options
+          r_build_args: '--no-build-vignettes --no-manual',
+          r_check_args: '--no-build-vignettes --no-manual --as-cran',
+          r_check_cran_incoming: false,
+          r_check_revdep: false,
+          # Heavy dependencies
+          latex: false,
+          pandoc: false,
+          pandoc_version: '1.12.4.2',
+          # Bioconductor
+          bioc: 'http://bioconductor.org/biocLite.R',
+          bioc_required: false,
+          bioc_use_devel: true,
```

Comment:
I think the default of `bioc_use_devel: true` might require some thought. Bioconductor's release cycle means that, for example, the current Bioc-devel can only be used with R 3.2 and not with R 3.1.x (or earlier). 

E.g. on R 3.1.2:

```r
> source("http://bioconductor.org/biocLite.R")
Installing package into ‘/Users/Peter/Desktop/bioc-devel_3.1.2/packrat/lib/x86_64-apple-darwin13.4.0/3.1.2’
(as ‘lib’ is unspecified)
trying URL 'http://www.bioconductor.org/packages/3.0/bioc/bin/macosx/mavericks/contrib/3.1/BiocInstaller_1.16.1.tgz'
Content type 'application/x-gzip' length 49535 bytes (48 Kb)
opened URL
==================================================
downloaded 48 Kb


The downloaded binary packages are in
	/var/folders/ns/tvfcs46s5b1f0k9bbl_q2k580000gn/T//RtmpK2ZqOJ/downloaded_packages
Bioconductor version 3.0 (BiocInstaller 1.16.1), ?biocLite for help
> useDevel()
Error: 'devel' version requires a more recent R
> sessionInfo()
R version 3.1.2 (2014-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BiocInstaller_1.16.1

loaded via a namespace (and not attached):
[1] packrat_0.4.1-1 tools_3.1.2 
`

Versus on R 3.2.0 (note also that you __can't__ use Bioc-release on 3.2 but only Bioc-devel)

```r
> source("http://bioconductor.org/biocLite.R")
Installing package into ‘/Users/Peter/Desktop/bioc-devel_3.2.0/packrat/lib/x86_64-apple-darwin13.4.0/3.2.0’
(as ‘lib’ is unspecified)
trying URL 'http://www.bioconductor.org/packages/3.1/bioc/bin/macosx/mavericks/contrib/3.2/BiocInstaller_1.17.5.tgz'
Content type 'application/x-gzip' length 51621 bytes (50 KB)
==================================================
downloaded 50 KB


The downloaded binary packages are in
	/var/folders/ns/tvfcs46s5b1f0k9bbl_q2k580000gn/T//Rtmp85NfXI/downloaded_packages
Bioconductor version 3.1 (BiocInstaller 1.17.5), ?biocLite for help
> 
> useDevel()
Error: 'devel' version already in use
> useDevel(FALSE)
Error: 'devel' version cannot be down-graded with this version of R
> sessionInfo()
R Under development (unstable) (2015-02-12 r67812)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.2 (Yosemite)

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BiocInstaller_1.17.5

loaded via a namespace (and not attached):
[1] tools_3.2.0     packrat_0.4.1-1
`

Bioconductor developers are encouraged to develop against Bioc-devel rather than Bioc-release, so I guess the issues facing Bioconductor developers wanting to use r-travis are:

1. Will it be possible to use R-devel (3.2, in this case) with r-travis? I'm currently using r-builder by @gaborcsardi (https://github.com/metacran/r-builder) in place of r-travis, but it would seem nice to consolidate this within the "official" Travis branch.
2. Can the `bioc_use_devel` flag be sanity checked for compability against the version of R requested by the user?